cmd/zpool/zpool_main.c

```diff
@@ -2209,34 +2209,29 @@ do_import(nvlist_t *config, const char *newname, const char *mntopts,
 		(void) fprintf(stderr, gettext("cannot import '%s': pool "
 		    "is formatted using an unsupported ZFS version\n"), name);
 		return (1);
-	} else if (state != POOL_STATE_EXPORTED &&
+	} else if (zfs_forceimport_required(config) && 
 	    !(flags & ZFS_IMPORT_ANY_HOST)) {
 		uint64_t hostid = 0;
-		unsigned long system_hostid = get_system_hostid();
+		char *hostname;
+		uint64_t timestamp;
+		time_t t;
 
 		(void) nvlist_lookup_uint64(config, ZPOOL_CONFIG_HOSTID,
 		    &hostid);
```

Comment:
zfs_force_import_required() may return B_TRUE even if ZPOOL_CONFIG_HOSTID is not present, (e.g. hostid == 0, I create pool, then reboot the machine, then set hostid, then import).

I agree WRT the keys already wrapped in verify().

What is the purpose of creating the extra local, t?  Might asctime() change it?  And why is "use '-f' to import any" a separate fprintf call?